### PR TITLE
ITS-3375 Adding a check so the idData connector data doesn't call the IDservice when there is no NINO from PDV.

### PR DESCRIPTION
**Pull request by rishisachania**
…ice when there is no NINO from PDV

### CODE REVIEW

File: app/connectors/IndividualDetailsConnector.scala [hunk 1]
```diff
@@ -47,19 +47,24 @@ class DefaultIndividualDetailsConnector @Inject() (httpClient: HttpClient,
...
-      additionalLogInfo
-    ) { implicit httpReads =>
-      EitherT(
-        httpClient
-          .GET(url)(httpReads, desApiHeaders(appConfig.individualDetails), ec)
-          .recovered(logger, connectorName, metrics.defaultRegistry, additionalLogInfo)
-      )
+
+    if (identifier.value.isEmpty) {
+      IndividualDetailsResponseEnvelope(Left(ConnectorError(400, "Cannot call getIndividualDetails with an empty identifier")))
```
Comment (ajchilts) [inline app/connectors/IndividualDetailsConnector.scala:52]: I am assuming that this is handled by the caller and the block correctly routes based on the fact this was not matched?

Comment (ajchilts) [reply app/connectors/IndividualDetailsConnector.scala:52]: Also do we need to add a test for this?

Comment (ajchilts) [reply app/connectors/IndividualDetailsConnector.scala:52]: @rishisachania  @rambabu-posa  - the more I think about this the more I do not like this approach. This is not a connector error. It's not an error at all. It's simply that we do not want to attempt to make this call if we have no Nino. Triggering an error is misleading and I think we need a different approach. Let me think about this.

**Review by ajchilts — CHANGES_REQUESTED** [review]

**Comment by rambabu-posa** [comment]
@ajchilts  Is it good to handle it at first step in Controller?

**Comment by ajchilts** [comment]
@rambabu-posa  - I think we need to omit the call if we do not need to make it. This is proving difficult with the way that the code is written as it expects either an error of individuals details. When in reality it's OK to have non matched and no individuals details.

**Review by ajchilts — APPROVED** [review]

**Pull request merged**